src/protocol/attribution/mod.rs

```diff
@@ -36,49 +38,96 @@ where
             .await?)
 }
 
-async fn compute_stop_bit<F, C, S>(
+///
+/// Computes `SUM(credits[i] through credits[i + n])` where `n` is the number of "matching rows", as indicated by the `helper_bits`
+/// This result is saved as credits[i].
+///
+/// Helper bits should be a sharing of either `1` or `0` for each row, indicating if that row "matches" the row preceeding it.
+/// Stop Bits should be initialized to a sharing of `1`
+///
+/// ## Errors
+/// Fails if the multiplication protocol fails.
+///
+pub async fn do_the_binary_tree_thing<F, C, S>(
     ctx: C,
-    record_id: RecordId,
-    b_bit: &S,
-    sibling_stop_bit: &S,
-    first_iteration: bool,
-) -> Result<S, Error>
+    helper_bits: &[S],
```

Comment:
having a second thought on that - it would probably be possible to avoid using vectors here if we pass the right-sized slice? i.e. ask clients to generate helper bits with share of one added to the end?

Reply:
eh I just realized that you need two copies, so it does not really matter whether you take a vec or a slice